src/actions/userActions.js

```diff
@@ -0,0 +1,68 @@
+import { userConstants } from '../constants/userConstants'
+import { userService } from '../services/userService'
+import { alertActions } from './alertActions'
+
+export const userActions = {
+    register,
+    login,
+    logout,
+    registerComplited
+
+};
+
+function login({email,password}) {
+    return dispatch => {
+        dispatch(request({email}));
+
+        userService.login(email,password).then(
+            (user) => {
+                dispatch(success({email}));
+                
+            },
+            (error) => {
+                
+                dispatch(failure({error}));
+            }
+        )
+
+        function request(user) { return { type: userConstants.LOGIN_REQUEST, user } }
+        function success(user) { return { type: userConstants.LOGIN_SUCCESS, user } }
```

Comment:
break this function into new lines to increase readability.

Reply:
Done